parser/options.go

```diff
@@ -125,3 +126,15 @@ func EnableOptionalSyntax(optionalSyntax bool) Option {
 		return nil
 	}
 }
+
+// EnableVariadicOperatorASTs enables a compact representation of chained like-kind commutative
+// operators. e.g. `a || b || c || d` -> `call(op='||', args=[a, b, c, d])`
+//
+// The benefit of enabling variadic operators ASTs is a more compact representation deeply nested
+// logic graphs.
+func EnableVariadicOperatorASTs(varArgASTs bool) Option {
+	return func(opts *options) error {
+		opts.enableVariadicOperatorASTs = varArgASTs
```

Comment:
I'm not sure this is a syntax change. It does feel like a runtime version change though; however, that only appears on the checked.proto. We could move up the runtime version into source_info?

Reply:
I was originally thinking of a minor version bump, but what you suggest sounds better. I do think we need a way of signaling the AST change for the runtime.

Reply:
Since the runtimes can be updated to support either format equally well, I think I'll just make it impossible to enable until we've made the update elsewhere.